lib/ocpp/v201/smart_charging.cpp

```diff
@@ -436,6 +511,55 @@ std::vector<ChargingProfile> SmartChargingHandler::get_profiles() const {
     return all_profiles;
 }
 
+ClearChargingProfileResponse SmartChargingHandler::clear_profiles(const ClearChargingProfileRequest& request) {
+    ClearChargingProfileResponse response;
+
+    const auto profile_id = request.chargingProfileId;
+    const auto criteria = request.chargingProfileCriteria;
+
+    auto any_found_profile = false;
+    // iterate over all profiles
+    for (auto& [existing_evse_id, evse_profiles] : charging_profiles) {
+        for (auto it = evse_profiles.begin(); it != evse_profiles.end();) {
+            // K10.FR.04: logical AND between the filters, if not set, the filter does not apply
+            if (profile_matches_clear_criteria(*it, profile_id, criteria, existing_evse_id)) {
+                any_found_profile = true;
+                evse_profiles.erase(it);
+                this->database_handler->delete_charging_profile(it->id);
+            } else {
+                // At least one of the filters did not match
+                ++it;
+            }
+        }
+    }
+
+    if (any_found_profile) {
```

Comment:
Can this boolean be removed? Set the response to Unknown by default, and on line 526 set it to Accepted.